src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -198,25 +254,118 @@ public AutoCompletionCandidates doAutoCompleteJobName(@QueryParameter String val
          * Fills in the available promotion processes.
          */
         public ListBoxModel doFillProcessItems(@AncestorInPath Job defaultJob, @QueryParameter("jobName") String jobName) {
+            final ListBoxModel r = new ListBoxModel();
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null) {
+                return r;
+            }
             if (!defaultJob.hasPermission(Item.CONFIGURE) && defaultJob.hasPermission(Item.EXTENDED_READ)) {
-                return new ListBoxModel();
+                return r;
             }
             defaultJob.checkPermission(Item.CONFIGURE);
 
             AbstractProject<?,?> j = null;
-            if (jobName!=null)
-                j = Jenkins.getInstance().getItem(jobName,defaultJob,AbstractProject.class);
-
-            ListBoxModel r = new ListBoxModel();
+            if (jobName != null) {
+                // JENKINS-25011: also look for jobs in folders.
+                j = getItemByPath(jobName, defaultJob, AbstractProject.class);
+            }
+           
             if (j!=null) {
                 JobPropertyImpl pp = j.getProperty(JobPropertyImpl.class);
                 if (pp!=null) {
-                    for (PromotionProcess proc : pp.getActiveItems())
+                    for (PromotionProcess proc : pp.getActiveItems()) {
+                        // Note: why not list all items instead of active ones?
+                        // this would allow to configure the job even
+                        // if a promotion hasn't happened (yet).
                         r.add(new Option(proc.getDisplayName(),proc.getName()));
+                    }
                 }
             }
             return r;
         }
     }
-
+    
+    //TODO: get relative paths based on folders?
+    //TODO: move to a helper class and then to the Jenkins core
+    @CheckForNull
+    @SuppressWarnings("unchecked")
+    private static <T extends Item> T getItemByPath(@Nonnull String path, @CheckForNull Item currentItem, 
+            @Nonnull Class<T> type) {
```

Comment:
This is a snippet, which should be moved into Jenkins core's utility classes. We will also need a support of other Job types in this plugin later. That's why I've implemented a generic method